lib/trino-filesystem-gcs/src/main/java/io/trino/filesystem/gcs/GcsFileSystemConfig.java

```diff
@@ -38,6 +43,16 @@ public class GcsFileSystemConfig
     private String jsonKey;
     private String jsonKeyFilePath;
 
+    // Note: without uniform strategy we cannot retry idempotent operations.
+    // The trino-filesystem api does not violate the conditions for idempotency, see https://cloud.google.com/storage/docs/retry-strategy#java for details.
+    private Optional<RetryStrategy> retryStrategy = Optional.of(RetryStrategy.UNIFORM);
+
+    private Optional<Duration> initialRetryDelay = Optional.of(new Duration(10, TimeUnit.MILLISECONDS));
+    // Note: there is no benefit to setting this much higher as the rpc quota is 1x per second: https://cloud.google.com/storage/docs/retry-strategy#java
+    private Optional<Duration> maxRetryDelay = Optional.of(new Duration(1100, TimeUnit.MILLISECONDS));
```

Comment:
Why those are optionals?

Reply:
In case we wanted to use the defaults. Initially was thinking that would be better but to match the behavior of the gcs hadoop connector switched to using retries. Do you think it would be better to make them nullable or set explicitly? lmk, whatever works.

Reply:
I think making these optional and using the library defaults is confusing for users, as that's inconsistent with how other config defaults work. Also, it seems like unnecessary complexity.